src/modules/producer.js

```diff
@@ -170,12 +171,16 @@ class Producer {
       if (!msg) msg = null;
 
       this._connection.config.transport.info('bmq:producer', `[${queue}] > `, msg);
-
-      return this.checkRpc(queue, parsers.out(msg, options), options);
+      return this.channel.checkQueue(queue).then(() => {
```

Comment:
in short I think checkQueue does a channel close if there is no queue. did you see any problems like this?